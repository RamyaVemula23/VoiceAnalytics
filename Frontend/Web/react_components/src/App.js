import React from 'react';
import 'bootstrap/dist/css/bootstrap.css'
import SimplePaper from './components/material-components/SimpleCard';
import DateRangepicker from './components/DateRangePicker';
import DonutChart from './components/charts/DonutChart'
import BarChart from './components/charts/BarChart'
import LineChart from './components/charts/LineChart'
import WordCloud from './components/charts/WordCloud'

function App() {
  const barData = [
    {
      "name": "Greeting",
      "percentage": 50
    },
    {
      "name": "Feedback",
      "percentage": 40
    },
    {
      "name": "Thanking",
      "percentage": 30
    }
  ]
  
  const donutData = {Claim: 9, Enquiry: 20, Complaint:30, reservation:8}
  const colorData = ['#A9A9A9','#000000','#FFFF00','#00FF00','#00BFFF']

  const lineMargin = {top: 10, right: 30, bottom: 30, left: 60}
  const sampleData = require('./components/data/ChartData.json')
  const lineData = sampleData.dataArray

  const wordCloudData = sampleData.wordCloudArray

  return (
    <div className="container">
      <div className="row">
        <div className="col-4"><SimplePaper heading="Dhanush" value="100" /></div>
        <div className="col-8"><DateRangepicker/></div>
      </div>

      <div className="row">
        <div className="col-6"><DonutChart width = "300" height = "200" scale = "20" data = {donutData} colorData = {colorData} /></div>
        <div className="col-6"><BarChart width = "300" height = "200" scale = "20" data = {barData} /></div>
      </div>

      <div className="row">
        <div className="col-6"><LineChart width = "300" height = "200" margin = {lineMargin} data = {lineData} /></div>
      </div>
      <br />
      <div className="row">
        <WordCloud width = "1200" height = "500" data = {wordCloudData} colorData = {colorData} />
      </div>

    </div>


  );
}

export default App;
