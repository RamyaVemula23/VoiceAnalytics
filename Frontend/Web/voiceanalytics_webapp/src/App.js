import React, { Component } from 'react';
import DashBoard from './pages/DashBoard';
import ApolloClient from 'apollo-client';
import { ApolloProvider } from '@apollo/react-hooks';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { createHttpLink } from 'apollo-link-http';
import { onError } from "apollo-link-error";
import ErrorBoundary from './components/base-components/ErrorBoundary'

  // const httpLink = createHttpLink({ uri: 'https://graphql-app.azurewebsites.net/graphql-server?code=AqxqBJsnF3BF8ERzZruUpUx8qeGMwR8jAq0M6wDWomGhfrRxp7hbiA==' })
  const httpLink = createHttpLink({ uri: 'http://localhost:7071/graphql-server' })

  const errorLink = onError(({ graphQLErrors, networkError }) => {
    if (graphQLErrors) {
      graphQLErrors.forEach(({ message, locations, path }) => {
        console.log(`[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`)
      }); 
    }
    else if (networkError) { 
      console.log(`[Network error]: ${networkError}`);
    } 
  });

  const client = new ApolloClient({
    link: errorLink.concat(httpLink),
    cache: new InMemoryCache()
  });

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { }
  }

  render() { 
    return ( 
      <React.Fragment>
        <ErrorBoundary>
        <ApolloProvider client={client}>
          <DashBoard />
        </ApolloProvider>
        </ErrorBoundary>
      </React.Fragment>
     );
  }
}
 
export default App;