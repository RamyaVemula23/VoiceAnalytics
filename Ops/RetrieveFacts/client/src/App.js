import React, { Component } from 'react';
import './App.css';
import CallTable from './components/CallTable';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      data : []
     }
  }

  componentDidMount() {
            fetch("http://localhost:4000/api/getFactsTable")
                .then(response => response.json())
                .then(res => {
                  this.setState({
                    data : res
                  })
                })
                
        } 

  
  render() { 
    console.log(this.state)
    let {data} = this.state
    return ( 
      <div>
        {data.length === 0 ?<p>Please wait while the data loads</p> :
        <CallTable keys = {Object.keys(data[0])} data = {data} />}
      </div>
     );
  }
}
 
export default App;
