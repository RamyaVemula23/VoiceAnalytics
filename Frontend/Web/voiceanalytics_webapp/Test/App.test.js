import React from 'react';
import ReactDOM from 'react-dom';
import {cleanup}from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from '../../voiceanalytics_webapp/src/App';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot",()=>{
const tree=renderer.create(<App/>).toJSON(); //convert in to virtual dom object
expect(tree).toMatchSnapshot();
});

