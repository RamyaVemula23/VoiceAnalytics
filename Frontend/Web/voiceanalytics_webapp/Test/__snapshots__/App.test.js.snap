// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    class="topnav"
  >
    <div>
      <nav
        className="navbar navbar-expand-lg navbar-dark bg-black"
      >
        <a
          href="https://www.dxc.technology/"
        >
          <img
            alt="website-logo"
            className="navbar-brand"
            src="dxc_logo.png"
            width="213px"
          />
        </a>
        <div
          className="collapse navbar-collapse"
          id="navbarSupportedContent"
        >
          <ul
            className="navbar-nav mr-auto"
          >
            <li
              className="nav-item custom-font-nav"
            >
              <a
                className="nav-link active"
                href="/dashboard"
                id="dashboard"
              >
                DASHBOARD
              </a>
            </li>
            <li
              className="nav-item custom-font-nav"
            >
              <a
                className="nav-link"
                href="/#"
                id="call-analytics"
              >
                CALL ANALYTICS
              </a>
            </li>
          </ul>
        </div>
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              style={
                Object {
                  "fill": "white",
                  "height": "50px",
                  "width": "50px",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
      </nav>
    </div>
     
  </div>,
  <div
    className="custom-container"
  >
    <div
      className="row"
    >
      <div
        className="test col-12"
      >
        Date Picker
      </div>
    </div>
    <div
      className="row"
    >
      <p>
        Loading...
      </p>
    </div>
    <div
      className="row"
    >
      <div
        className="test col-12"
      >
        Filter
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-4 col-md-12 col-sm-12 col-12"
      >
        <p>
          Loading...
        </p>
      </div>
      <div
        className="col-lg-4 col-md-12 col-sm-12 col-12"
      >
        <p>
          Loading...
        </p>
      </div>
      <div
        className="col-lg-4 col-md-12 col-sm-12 col-12"
      >
        <p>
          Loading...
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <p>
          Loading...
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 card-heading"
      >
        KEYWORDS ON CALL
      </div>
      <div
        className="col-12"
      >
        <p>
          Loading...
        </p>
      </div>
    </div>
  </div>,
  <footer
    className="footer mt-auto py-3 bg-black text-light"
  >
    <div
      className="custom-footer-container row"
    >
      <div
        className="col-6"
      >
        Privacy policy
      </div>
      <div
        className="col-6 text-right"
      >
        Â©DXC Technology
      </div>
    </div>
  </footer>,
]
`;
